index.js

```diff
@@ -12,8 +17,15 @@ function init() {
 
 	const selectedText = editor.getSelectedText();
 	const text = selectedText || editor.getText();
+	const config = atom.config.get('perfectionist');
+
+	config.syntax = 'scss';
 
-	postcss(perfectionist(atom.config.get('perfectionist'))).process(text, {
+	if (editor.getGrammar().scopeName === 'source.css') {
+		delete config.syntax;
+	}
```

Comment:
You were not. This is just a different version of the above. In https://github.com/sindresorhus/atom-perfectionist/pull/7/commits/2d21b7fc2bd9944b5239d77153402f88a3baa41b#diff-168726dbe96b3ce427e7fedce31bb0bcR23, you were setting the `syntax` option when it's a SCSS file. This is setting the SCSS when it's **not** a CSS file. That's an important difference and what makes it work for SCSS on unsaved buffers.